HotFix/Transport/TcpTransport.cs

```diff
@@ -33,17 +34,32 @@ public static TcpTransport Initiate(IClock clock, string address, int port)
         public static TcpTransport Accept(IClock clock, string address, int port)
         {
             var endpoint = new IPEndPoint(IPAddress.Parse(address), port);
-            var socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp) { ReceiveTimeout = 1000, NoDelay = true };
-
-            socket.EnableFastPath();
-            socket.Bind(endpoint);
-            socket.Listen(1);
-
-            var client = socket.Accept();
-
-            socket.Dispose();
-
-            return new TcpTransport(clock, client);
+            using (var socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp) { ReceiveTimeout = 1000, NoDelay = true })
+            {
+                socket.EnableFastPath();
+                socket.Bind(endpoint);
+                socket.Listen(1);
+
+                Socket client = null;
+                var e = new SocketAsyncEventArgs();
+                e.Completed += (sender, args) => client = args.AcceptSocket;
+
+                if (!socket.AcceptAsync(e))
+                {
+                    client = e.AcceptSocket;
+                }
+                else
+                {
+                    for (var i = 0; i < 20; i++)
+                    {
+                        if (client != null) break;
+                        Thread.Sleep(500);
```

Comment:
Would it not be better to pass the cancellation token down here and check that in a never ending loop? That way we wouldn't have to count time or assume 10 seconds but rather just call `EndAccept` if cancelled etc...